src/app/data-model-provider/MetadataTypes.h

```diff
@@ -84,18 +84,99 @@ enum class AttributeQualityFlags : uint32_t
     kTimed           = 0x0040, // `T` quality on attributes (writes require timed interactions)
 };
 
+
+
+
 struct AttributeEntry
 {
     AttributeId attributeId;
-    BitFlags<AttributeQualityFlags> flags;
 
-    // read/write access will be missing if read/write is NOT allowed
-    //
-    // NOTE: this should be compacted for size
-    std::optional<Access::Privilege> readPrivilege;  // generally defaults to View if readable
-    std::optional<Access::Privilege> writePrivilege; // generally defaults to Operate if writable
+    // Constructor
+    constexpr AttributeEntry(AttributeId id = 0, // attributeId initial value
+                             std::underlying_type_t<AttributeQualityFlags> attrQualityFlags = 0, // mask.flags initial value
+                             std::underlying_type_t<Access::Privilege> readPriv = kNoPrivilege, // mask.readPrivilege initial value
```

Comment:
why do we use std::underlying_type_t in the signature instead of the forced enum type (or the bitflags version for the flags)?